blog/user_roles.md

```diff
@@ -0,0 +1,294 @@
+first post I briefly mentioned that I was going to add the user under the accounts context. Now, we currently don't have any contexts setup, so let's take advantage
+of Phoenix generators to set it up after we've alread created the user. I have confidence that Dan has taken care of the basic testing of the basic user level
+fields and changeset, so we are mainly concerned with testing our customisations. Testing the API layer(ie. the context layer) is a very nice
+way to test actual behaviour. But can we actually run the generators when we already have part of the contet defined? Let's find out:
+
+```bash
+mix phx.gen.context Accounts User users
+
+# outputs:
+
+The following files conflict with new files to be generated:
+
+  * lib/student_manager/accounts/user.ex
+
+See the --web option to namespace similarly named resources
+
+Proceed with interactive overwrite? [Yn] 
+* creating lib/student_manager/accounts/user.ex
+lib/student_manager/accounts/user.ex already exists, overwrite? [Yn] n
+* creating priv/repo/migrations/20190616212819_create_users.exs
+* creating lib/student_manager/accounts.ex
+* injecting lib/student_manager/accounts.ex
+* creating test/student_manager/accounts_test.exs
+* injecting test/student_manager/accounts_test.exs
+
+Remember to update your repository by running migrations:
+
+    $ mix ecto.migrate
+```
+
+Now that is pretty awesome actually. Once we go into `interactive overwrite` it'll ask us if we want to replace any existing files. In our case that's only the `user.ex`
+file, and we don't want to override that one! You can also see that it created a new migration, but as we already have that in place we can simply remove that file.
+
+Now this gives us a starting point to start testing our `accounts` context. We'll actually remove some of the boilerplate tests as we don't need them right now and
+keep a subset we want to test now. This is what is left of the tests:
+```elixir
+defmodule StudentManager.AccountsTest do
+  use StudentManager.DataCase
+
+  alias StudentManager.Accounts
+
+  describe "users" do
+    alias StudentManager.Accounts.User
+
+    @valid_attrs %{email: "email@example.com", password: "supersecretpassword", confirm_password: "supersecretpassword"}
+    @invalid_attrs %{email: "bademail@bad", password: "short", confirm_password: "shot" }
+
+    def user_fixture(attrs \\ %{}) do
+      {:ok, user} =
+        attrs
+        |> Enum.into(@valid_attrs)
+        |> Accounts.create_user()
+
+      user
+    end
+
+    test "create_user/1 with valid data creates a user" do
+      assert {:ok, %User{} = user} = Accounts.create_user(@valid_attrs)
+    end
+
+    test "create_user/1 with invalid data returns error changeset" do
+      assert {:error, %Ecto.Changeset{}} = Accounts.create_user(@invalid_attrs)
+    end
+
+    test "create_user/1 creates a user with a default role of student" do
+      {:ok, user} = Accounts.create_user(@valid_attrs)
```

Comment:
I'm not sure this belongs here? It's not the responsibility of the `Accounts` module, but the `Account` module, right?

Reply:
Well it is not the `Accounts module`, but rather the `Accounts context` that is the API layer responsible for any interaction related to accounts. Inside the Accounts context we have access to users, parents, invoicing etc. Maybe not precisely these, but that is the idea behind this. The Accounts context is actually what they use in the Phoenix book.